Formula/pyside.rb

```diff
@@ -11,38 +11,39 @@ class Pyside < Formula
   end
 
   depends_on "cmake" => [:build, :test]
-  depends_on "llvm"
+  depends_on "llvm" => :build
   depends_on "python"
   depends_on "python@2"
   depends_on "qt"
 
   def install
+    ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
+
     args = %W[
-      --no-user-cfg
-      install
-      --prefix=#{prefix}
-      --install-scripts=#{bin}
-      --single-version-externally-managed
-      --record=installed.txt
       --ignore-git
       --parallel=#{ENV.make_jobs}
+      --install-scripts #{bin}
     ]
 
     xy = Language::Python.major_minor_version "python3"
 
-    system "python3", "setup.py", *args,
-           "--install-lib", lib/"python#{xy}/site-packages"
+    system "python3", *Language::Python.setup_install_args(prefix),
+           "--install-lib", lib/"python#{xy}/site-packages", *args
 
     lib.install_symlink Dir.glob(lib/"python#{xy}/site-packages/PySide2/*.dylib")
     lib.install_symlink Dir.glob(lib/"python#{xy}/site-packages/shiboken2/*.dylib")
 
-    system "python2", "setup.py", *args,
-           "--install-lib", lib/"python2.7/site-packages"
+    rm_rf bin/"shiboken2"
```

Comment:
If it is installed by `setup.py` by default, we probably don't want to remove it.

Reply:
The shiboken binary is not a part of `pyside` and is not required for using the binding. Indeed, the official distribution of `pyside` does not include it. The reason why it was not removed before was to allow a test of the correct linkage of `llvm`, which was opportunistically linked in a way that it was not possible to avoid, until now.

Reply:
I do not understand:

> The shiboken binary is not a part of pyside

and 

> the official distribution of pyside does not include it

We are using the official source distribution, are we not? It appears to be included, and installed by default. If this is not the desired behaviour, it should be reported upstream as a bug.

Homebrew's policy is that we want to ship things as close to possible as the default building configuration. Every deviation from that is possible cause for breakage and extra maintenance burden down the road.

Reply:
The official installation method is based on pre-built wheels loaded from pypi, via running `pip install PySide2`. You can verify that installations done via this method do not include a shiboken2 nor a `libclang.dylib` binary. 

Installing from the source archive creates and installs a shiboken2 binary, instead, essentially because that binary is required to build pyside, but not to run it. That shiboken2 binary is linked to libclang (from `llvm`) and so it was removed in this formula.

Reply:
If you are particularly concerned about the manual removals, I just realized and tested that it is possible to achieve the same result by running 
```
    system "python3", "setup.py", *args,
           "--install-lib", lib/"python#{xy}/site-packages",
           "--build-type=shiboken2"
           
    system "python3", "setup.py", *args,
           "--install-lib", lib/"python#{xy}/site-packages",
           "--build-type=pyside2"
```
for each python version, instead of the previous single `setup.py` invocation.

Does this sound any better?

Reply:
> Does this sound any better?

Yes, please!